depthai_nodes/ml/parsers/detection.py

```diff
@@ -0,0 +1,137 @@
+        @param output_layer_name: The name of the output layer(s) from which the scores are extracted.
+        @type output_layer_name: Union[str, List[str]]
+        @param conf_threshold: Confidence score threshold of detected bounding boxes.
+        @type conf_threshold: float
+        @param iou_threshold: Non-maximum suppression threshold.
+        @type iou_threshold: float
+        @param max_detections: Maximum number of detections to keep.
+        @type max_detections: int
+
+        Output Message/s
+        -------
+        **Type**: ImgDetectionsExtended
+
+        **Description**: ImgDetectionsExtended message containing bounding boxes and confidence scores of detected objects.
+        """
+        super().__init__()
+        self.output_layer_name = output_layer_name
+        self.conf_threshold = conf_threshold
+        self.iou_threshold = iou_threshold
+        self.max_detections = max_detections
+
+    def setOutputLayerName(self, output_layer_name: str) -> None:
+        """Sets the output layer name(s) for the parser.
+
+        @param output_layer_name: The name of the output layer(s) from which the scores
+            are extracted.
+        @type output_layer_name: Union[str, List[str]]
+        """
+        self.output_layer_name = output_layer_name
+
+    def setConfidenceThreshold(self, threshold: float) -> None:
+        """Sets the confidence score threshold for detected objects.
+
+        @param threshold: Confidence score threshold for detected objects.
+        @type threshold: float
+        """
+        self.conf_threshold = threshold
+
+    def setIOUThreshold(self, threshold: float) -> None:
+        """Sets the non-maximum suppression threshold.
+
+        @param threshold: Non-maximum suppression threshold.
+        @type threshold: float
+        """
+        self.iou_threshold = threshold
+
+    def setMaxDetections(self, max_det: int) -> None:
+        """Sets the maximum number of detections to keep.
+
+        @param max_det: Maximum number of detections to keep.
+        @type max_det: int
+        """
+        self.max_det = max_det
+
+    def build(self, head_config) -> "DetectionParser":
+        """Sets the head configuration for the parser.
+
+        Attributes
+        ----------
+        head_config : Dict
+            The head configuration for the parser.
+
+        Returns
+        -------
+        DetectionParser
+            Returns the parser object with the head configuration set.
+        """
+
+        output_layer = head_config["outputs"]
+        if len(output_layer) != 1:
+            raise ValueError(
+                f"Only one output layer supported for Detection, got {output_layer} layers."
```

Comment:
This is not true in many cases: Yunet has more layers, SCRFD has more layers, Palm detection has more layers